src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs

```diff
@@ -1178,36 +1181,36 @@ internal static StackValue[] GenerateRangeByAmount(decimal start, long amount, d
                        Resources.kInvalidAmountInRangeExpression);
                     return null;
                 }
-                decimal stepsize = new decimal(svStep.ToDouble().RawDoubleValue);
+                decimal stepsize = new decimal(svStep.ToDouble().DoubleValue);
                 return GenerateRangeByAmount(start, amount, stepsize, isIntRange);
             }
             else
             {
-                switch (svOp.opdata)
+                switch (svOp.IntegerValue)
```

Comment:
Seems like casting this to `RangeStepOperator` will avoid casting all the constants in this `switch-case` statement to `int` ;)